presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -621,16 +637,21 @@ else if (hiveType.equals(HIVE_LONG)) {
     {
         private final Map<String, Partition> partitions;
         private final Set<String> prunedPartitionNames;
+        private final Map<String, Set<ColumnHandle>> redundantColumnDomains;
```

Comment:
While reviewing the 2nd commit, maybe it's better to model these objects as`Map<String, PartitionSplitInfo>`, where PartitionSplitInfo (or whatever a better name) is an encapsulation. It will have Partition, boolean-typed pruned flag, and a set of redundantColumnDomains.